src/details/ArborX_DetailsCrsGraphWrapperImpl.hpp

```diff
@@ -210,6 +210,13 @@ void queryImpl(ExecutionSpace const &space, Tree const &tree,
 
   int const n_results = KokkosExt::lastElement(space, offset);
 
+  if(n_results < 0){
+    // overflow
+    throw std::runtime_error("The search resulted in too many neighbor pairs. "
+                             "Please ensure the primitive positions are not "
+                             "in a degenerate state.");
+  }
```

Comment:
Maybe, just
```suggestion
  // Catch possible overflow
  ARBORX_ASSERT(n_results >= 0);
```

Reply:
This is more helpful than the current state. Though as a user of this library (or for that matter any library), I would always prefer to avoid stepping through the library code to find out what an internal variable does.